spec/services/incremental_refresh_upload_task_service_spec.rb

```diff
@@ -14,15 +13,38 @@
       allow(ClowderConfig).to receive(:instance).and_return({"UPLOAD_URL" => "http://www.upload_url.com"})
     end
 
-    it "returns IncrementalRefreshUploadTask" do
-      task = subject.process.task
+    context "source has no sha and size" do
+      let(:source) { FactoryBot.create(:source, :tenant => tenant) }
 
-      expect(task.type).to eq('IncrementalRefreshUploadTask')
-      expect(task.input["response_format"]).to eq('tar')
-      expect(task.input["jobs"].count).to eq(12)
-      expect(task.input["upload_url"]).to eq('http://www.upload_url.com')
-      expect(task.state).to eq('pending')
-      expect(task.status).to eq('ok')
+      it "returns IncrementalRefreshUploadTask" do
+        task = subject.process.task
+
+        expect(task.type).to eq('IncrementalRefreshUploadTask')
+        expect(task.input["response_format"]).to eq('tar')
+        expect(task.input["jobs"].count).to eq(12)
+        expect(task.input["upload_url"]).to eq('http://www.upload_url.com')
+        expect(task.input["previous_sha"]).to be_nil
```

Comment:
@hsong-rh Should we check if the key doesn't exist here.

Reply:
Use `task.input.keys` to check valid keys